### PR TITLE
feature: update to laravel 10

### PR DESCRIPTION
**Pull request by taka-oyama**
Fixed
- `Connection::reconnectIfMissingConnection` was changed from `protected` to `public` to match laravel 10.
- [Query/Expression](https://laravel.com/docs/10.x/upgrade#database-expressions) changed from `(string)$expr`to `$expr->toValue($grammar)`.
- Applied [QueryException constructor change](https://laravel.com/docs/10.x/upgrade#query-exception-constructor) to `Schema/Grammar`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # v5.0.0 (Not Released Yet)
 
+updated composer.json to only support laravel 10
+
+Fixed
+- `Connection::reconnectIfMissingConnection` was changed from `protected` to `public` to match laravel 10.
+- [Query/Expression](https://laravel.com/docs/10.x/upgrade#database-expressions) changed from `(string)$expr`to `$expr->toValue($grammar)`.
```
Comment (tyabu12) [inline CHANGELOG.md:7]:
```suggestion
- [Query/Expression](https://laravel.com/docs/10.x/upgrade#database-expressions) changed from `(string)$expr` to `$expr->getValue($grammar)`.
```

**Review by tyabu12 — CHANGES_REQUESTED** [review]

**Review by zeriyoshi — APPROVED** [review]

File: src/Connection.php [hunk 2]
```diff
@@ -450,7 +455,7 @@ protected function runQueryCallback($query, $bindings, Closure $callback)
      */
     protected function getSessionNotFoundMode(): string
     {
-        return $this->config['sessionNotFoundErrorMode'] ?? self::MAINTAIN_SESSION_POOL;
+        return $this->config['sessionNotFoundErrorMode'] ?? self::CLEAR_SESSION_POOL;
```
Comment (halnique) [inline src/Connection.php:458]: The [README.md](https://github.com/colopl/laravel-spanner/tree/feature/update-laravel-10#session-not-found-exception-handling) should also be modified accordingly.
It is my understanding that the `default` value changes.

File: src/Schema/Blueprint.php [hunk 3]
```diff
@@ -142,11 +142,13 @@ public function floatArray($column)
...
     /**
      * @param string $column
-     * @param int|string $length
+     * @param int|string|null $length
      * @return ColumnDefinition
      */
-    public function stringArray($column, $length)
+    public function stringArray($column, $length = null)
     {
+        $length = $length ?: Builder::$defaultStringLength;
```
Comment (halnique) [inline src/Schema/Blueprint.php:150]:
```suggestion
        $length ??= Builder::$defaultStringLength;
```

File: src/Connection.php (hunk 2 above)
Comment (taka-oyama) [reply src/Connection.php:458]: Thank you. README.md has been fixed.

**Review by halnique — APPROVED** [review]

**Review by zeriyoshi — APPROVED** [review]

**Pull request merged**